pandas/io/pytables.py

```diff
@@ -3469,18 +3459,29 @@ def f(i, c):
 
             atom = getattr(desc, c)
             adj_name = _maybe_adjust_name(c, self.version)
+
+            # TODO: why kind_attr here?
+            values = getattr(table_attrs, f"{adj_name}_kind", None)
+            dtype = getattr(table_attrs, f"{adj_name}_dtype", None)
+            kind = _dtype_to_kind(dtype)
+
             md = self.read_metadata(c)
-            meta = "category" if md is not None else None
+            # TODO: figure out why these two versions of `meta` dont always match.
```

Comment:
what is going on here?

Reply:
So ever place where we set `self.meta` it is equivalent to `meta = "category" if md is not None else none`, but in this one place where we read it from the file, there is one test case where it doesnt match, and i dont know why.

if/when that is resolved, `meta` can be made a property